server/server.go

```diff
@@ -332,18 +332,24 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		// Disable caching of responses.
 		w.Header().Set("Cache-control", "no-cache")
 
-		ae := r.Header.Get(util.AcceptEncodingHeader)
-		switch {
-		case strings.Contains(ae, util.SnappyEncoding):
-			w.Header().Set(util.ContentEncodingHeader, util.SnappyEncoding)
-			s := newSnappyResponseWriter(w)
-			defer s.Close()
-			w = s
-		case strings.Contains(ae, util.GzipEncoding):
-			w.Header().Set(util.ContentEncodingHeader, util.GzipEncoding)
-			gzw := newGzipResponseWriter(w)
-			defer gzw.Close()
-			w = gzw
+		// Don't try to compress responses to the /debug/{events,requests,etc}
+		// paths. Doing so fouls up the content-type detection resulting in a
```

Comment:
Applied this diff locally:
```diff
diff --git a/trace/trace.go b/trace/trace.go
index c44cb7e..10734a1 100644
--- a/trace/trace.go
+++ b/trace/trace.go
@@ -113,6 +113,7 @@ func init() {
 			http.Error(w, "not allowed", http.StatusUnauthorized)
 			return
 		}
+		w.Header().Set("Content-Type", "text/html")
 		Render(w, req, sensitive)
 	})
 	http.HandleFunc("/debug/events", func(w http.ResponseWriter, req *http.Request) {
@@ -121,6 +122,7 @@ func init() {
 			http.Error(w, "not allowed", http.StatusUnauthorized)
 			return
 		}
+		w.Header().Set("Content-Type", "text/html")
 		RenderEvents(w, req, sensitive)
 	})
 }
```

I end up with a page that looks like this:
![screen shot 2016-01-04 at 1 32 52 pm](https://cloud.githubusercontent.com/assets/1535036/12097086/af2ef3ac-b2e7-11e5-96b8-f0eb6edb5f4c.png)

Any theories?